src/smt/smt_setup.cpp

```diff
@@ -781,7 +785,6 @@ namespace smt {
     }
 
     void setup::setup_str_noodler() {
-        setup_arith();
```

Comment:
Can you comment on this please?

Reply:
I moved it to line 573, in `setup_QF_S()`. This is because if we do not set any logic, I think Z3 calls `setup_unknown()`, in which arith theory is registered and string theories are registered in `setup_seq_str()`, where I want to register just Noodler, and not arith theory again.

But this whole file is very weird, everything is very ad-hoc.